File: javascript-systempromptleakage/vulnerable14.js
const express = require('express')
const OpenAI = require('openai')

const app = express()
const openai = new OpenAI()

const tools = [
  { type: 'function', function: {
    name: 'lookup_internal_pricing',
    description: 'Returns confidential pricing tiers for enterprise accounts.',
    parameters: { type: 'object', properties: { tier: { type: 'string' } } }
  }}
]

app.post('/chat', async (req, res) => {
  await openai.chat.completions.create({
    model: 'gpt-4o',
    messages: [{ role: 'user', content: req.body.message }],
    tools,
  })
  res.json({ reply: 'ok' })
})

app.get('/debug/tools', (req, res) => {
  res.json({ tools })
})

app.listen(3000)

File: python-supplychain/vulnerable14.py
import subprocess
import openai
from fastapi import FastAPI

app = FastAPI()
client = openai.OpenAI()

@app.post("/install-package")
def install_package(user_request: str):
    response = client.chat.completions.create(
        model="gpt-4",
        messages=[{"role": "user", "content": f"What pip package should I use for: {user_request}? Reply with only the package name."}],
    )
    package_name = response.choices[0].message.content.strip()
    subprocess.run(["pip", "install", package_name], check=True)
    return {"installed": package_name}
